packages/app/helpers/ens.ts

```diff
@@ -33,52 +33,79 @@ const abi = [
   },
 ];
 
-export const useEnsLookup = (
-  addresses: string[],
-  signerOrProvider: any
-): (string | null)[] => {
-  const addResolvedAddresses = useEnsStore(
-    (state) => state.addResolvedAddresses
+export async function lookupEns(addresses: string[], contract: any) {
+  if (addresses.length === 0) {
+    return [];
+  }
+  const [allNames] = await contract.functions.getNames(addresses);
+  const validNames = allNames.map((n: any) => {
+    return namehash.normalize(n) === n && n !== "" ? n : null;
+  });
+  return validNames;
+}
+
+export const useEnsContract = () => {
+  const signerOrProvider = useProvider();
+  const contractConfig = useMemo(
+    () => ({
+      // mainnet ens contract for reverse lookup
+      addressOrName: "0x3671aE578E63FdF66ad4F3E12CC0c0d71Ac7510C",
+      contractInterface: abi,
+      signerOrProvider,
+    }),
+    [signerOrProvider]
   );
-  // check if the ens store has all of the addresses already resolved
-  const resolvedNames = useEnsStore((state) => state.resolvedNames);
+  const contract = useContract(contractConfig);
+  const contractRef = useRef(contract);
```

Comment:
using a ref since useContract seemed to return a new instance when it probably shouldn't have. I need to double check this now that i've memoized the config being passed in (see above)